src/main/java/io/logz/jmx2logzio/Jmx2Logzio.java

```diff
@@ -55,8 +55,15 @@ private long calcDurationInSecondsUntilNextPollingIntervalStartTime() {
 
     @Override
     public void shutdown() {
-        logger.info("Shutting down...");
+        logger.info("Requesting metrics poller to stop");
+        taskScheduler.shutdown();
+        try {
+            taskScheduler.awaitTermination(20, TimeUnit.SECONDS);
+        } catch (InterruptedException e) {
+            logger.warn("final request was interrupted");
+        }
         taskScheduler.shutdownNow();
```

Comment:
1) I feel there is no need for `shutdownNow` if we already do `shutdown` and `awaitTermination`
2) add e message to the warning log